### PR TITLE
Shall vs shall only

### LINKED ISSUE #314: If this key is present, cryptographic operations that use the symmetric card authentication key SHALL..
### All Fields Are Required
**Organization Name (N/A, if individual)**: Rick Uhrig

**Organization Type (see below for codes)**: 4 - Self
 
**Reference (Include section/paragraph or pdf line number)**: 4.2.2.3 Line 1874 - 1875
 
**Comment (Include rationale for comment)**: The statement "The cryptographic operations that use the card authentication key SHALL be available through the contact and contactless interfaces of the PIV Card." is silent on the virtual contact interface.

 
**Suggested Change**: Replace with "If this key is present, cryptographic operations that use the symmetric card authentication key SHALL be available through the contact, virtual contact, and contactless interfaces of the PIV Card."
 
---
 
Organization Type: 1 = Federal, 2 = Industry, 3 = Academia, 4 = Self, 5 = Other

### LINKED ISSUE #315: If this key is present, cryptographic operations using the PIV Card's digital signature key SHALL be available through the contact interface
### All Fields Are Required
**Organization Name (N/A, if individual)**: Rick Uhrig

**Organization Type (see below for codes)**: 4 - Self
 
**Reference (Include section/paragraph or pdf line number)**: 4.2.2.4 Line 1879-1881
 
**Comment (Include rationale for comment)**: "If this key is present, cryptographic operations using the digital signature key SHALL be performed using the contact and virtual contact interfaces of the PIV Card." does not require the key to be available on the contact or virtual contact interfaces.  This does not seem to be the intent.

 
**Suggested Change**: Replace with "If this key is present, cryptographic operations using the PIV Card's digital signature key SHALL be available through the contact interface, SHALL be available through the virtual contact interface, and SHALL NOT be available through the contactless interface."

 
---
 
Organization Type: 1 = Federal, 2 = Industry, 3 = Academia, 4 = Self, 5 = Other

### LINKED ISSUE #316: If this key is present, the cryptographic operations that use the PIV Card's key management key SHALL..
### All Fields Are Required
**Organization Name (N/A, if individual)**: Rick Uhrig

**Organization Type (see below for codes)**: 4 - Self
 
**Reference (Include section/paragraph or pdf line number)**: 4.2.2.5 Line 1890-1892
 
**Comment (Include rationale for comment)**: "If present, the cryptographic operations that use the key management key SHALL only be accessible using the contact and virtual contact interfaces of the PIV Card." does not require the key to be available on the contact or virtual contact interfaces.  This does not seem to be the intent.
 
**Suggested Change**: Replace with "If this key is present, the cryptographic operations that use the PIV Card's key management key SHALL be available through the contact interface, SHALL be available through the virtual contact interface, and SHALL NOT be available on the contactless interface."

 
---
 
Organization Type: 1 = Federal, 2 = Industry, 3 = Academia, 4 = Self, 5 = Other

### LINKED ISSUE #317: The PIV Card application administration key SHALL be available through the contact interface..
### All Fields Are Required
**Organization Name (N/A, if individual)**: Rick Uhrig

**Organization Type (see below for codes)**: 4 - Self
 
**Reference (Include section/paragraph or pdf line number)**: 4.2.2.6 Line 1900-1901
 
**Comment (Include rationale for comment)**: "If present, the cryptographic operations that use the PIV Card application administration key SHALL only be accessible using the contact interface of the PIV Card."

 
**Suggested Change**: Replace with "If present, the cryptographic operations that use the PIV Card application administration key SHALL be available through the contact interface, SHALL NOT be available through the virtual contact interface, and SHALL NOT be available on the contactless interface.""

 
---
 
Organization Type: 1 = Federal, 2 = Industry, 3 = Academia, 4 = Self, 5 = Other

### PR DESCRIPTION
**Pull request by hferraio**
closes #317 
closes #315 
closes #314 
closes #316

### CODE REVIEW

File: _FIPS201/frontend.md [hunk 1]
```diff
@@ -708,8 +708,8 @@ key management keys.
 #### 4.2.2.6 PIV Card Application Administration Key {#s-4-2-2-6}
 If present, the PIV Card application administration
 key **SHALL** be imported onto the card by the issuer. If present, the cryptographic operations that use
-the PIV Card application administration key **SHALL** only be accessible using the contact interface of
-the PIV Card.
+the PIV Card application administration key **SHALL** be available through the contact interface of
+the PIV Card and **SHALL NOT** be available through the virtual contact interface of the PIV Card.
```
Comment (jimfenton) [inline _FIPS201/frontend.md:712]:
```suggestion
the PIV Card and **SHALL NOT** be available through the virtual contact and contactless interfaces of the PIV Card.
```

**Review by jimfenton — CHANGES_REQUESTED** [review]
Suggestion for one place where we don't address all three interface types.

I'm puzzled, though, about why some of these allow contact and contactless but not virtual contact interfaces. Is there a rationale for that?

File: _FIPS201/frontend.md [hunk 2]
```diff
@@ -696,8 +696,8 @@ PIV digital signature keys.
 
 #### 4.2.2.5 Key Management Key {#s-4-2-2-5}
 This key **MAY** be generated on the PIV Card or imported to the card. If
-present, the cryptographic operations that use the key management key **SHALL** only be accessible using
-the contact and virtual contact interfaces of the PIV Card. Private key operations **MAY** be
+present, the cryptographic operations that use the key management key **SHALL** available through
```
Comment (hferraio) [inline _FIPS201/frontend.md:699]:
```suggestion
present, the cryptographic operations that use the key management key **SHALL** be available through 
```

File: _FIPS201/frontend.md [hunk 3]
```diff
@@ -635,8 +635,8 @@ management requirements are also specified.
 #### 4.2.2.1 PIV Authentication Key {#s-4-2-2-1}
 This key **SHALL** be generated on the PIV Card. The PIV Card **SHALL NOT**
 permit exportation of the PIV authentication key. The cryptographic operations that use the PIV
-authentication key **SHALL** be available only through the contact and virtual contact interfaces of the
-PIV Card. Private key operations **MAY** be performed using an activated PIV Card without explicit
+authentication key **SHALL** be available through the contact and virtual contact interfaces of the
```
Comment (regenscheid) [inline _FIPS201/frontend.md:638]: Being somewhat pedantic, should this be "SHALL" or should it instead be "MAY only"?

Or, perhaps more accurately, "SHALL be available over the contact interface and MAY be available over the virtual contact interface".  You wouldn't want a card that can't do PIV auth over the contact interface, but you can have one that doesn't have a VCI.

Comment (regenscheid) [reply _FIPS201/frontend.md:638]:
```suggestion
authentication key **SHALL** be available through the contact interface and **MAY** additionally be available over the virtual contact interface of the
```

File: _FIPS201/frontend.md [hunk 4]
```diff
@@ -635,8 +635,8 @@ management requirements are also specified.
...
+PIV Card and **SHALL NOT** be available through the contactless interface of the PIV Card.  Private key operations **MAY** be performed using an activated PIV Card without explicit
```
Comment (regenscheid) [inline _FIPS201/frontend.md:639]:
```suggestion
PIV Card. The PIV authentication key **SHALL NOT** be available through the contactless interface of the PIV Card.  Private key operations **MAY** be performed using an activated PIV Card without explicit
```

File: _FIPS201/frontend.md [hunk 5]
```diff
@@ -675,15 +675,15 @@ onto the card by the issuer or be generated on the card. If present, the symmetr
...
+contactless interfaces of the PIV Card and **SHALL NOT** be available through the virtual contact interface of the PIV Card. This Standard does not specify key management protocols or
 infrastructure requirements.
 
 #### 4.2.2.4 Digital Signature Key {#s-4-2-2-4}
 The PIV digital signature key **SHALL** be generated on the PIV Card. The PIV
 Card **SHALL NOT** permit exportation of the digital signature key. If this key is present, cryptographic operations
-using the digital signature key **SHALL** be performed using the contact and virtual contact
-interfaces of the PIV Card. Private key operations **SHALL NOT** be performed without explicit user action,
+that use the digital signature key **SHALL** be available through the contact and virtual contact
+interfaces of the PIV Card and **SHALL NOT** be available through the contactless interface of the PIV Card. Private key operations **SHALL NOT** be performed without explicit user action,
```
Comment (regenscheid) [inline _FIPS201/frontend.md:686]:
```suggestion
that use the digital signature key **SHALL** be available through the contact interface and **MAY** additionally be available over the virtual contact
interface of the PIV Card. Operations that use the digital signature key **SHALL NOT** be available through the contactless interface of the PIV Card. Private key operations **SHALL NOT** be performed without explicit user action,
```

File: _FIPS201/frontend.md [hunk 6]
```diff
@@ -696,8 +696,8 @@ PIV digital signature keys.
 
 #### 4.2.2.5 Key Management Key {#s-4-2-2-5}
 This key **MAY** be generated on the PIV Card or imported to the card. If
-present, the cryptographic operations that use the key management key **SHALL** only be accessible using
-the contact and virtual contact interfaces of the PIV Card. Private key operations **MAY** be
+present, the cryptographic operations that use the key management key **SHALL** be available through 
+the contact and virtual contact interfaces of the PIV Card and **SHALL NOT** be available through the contactless interface of the PIV Card. Private key operations **MAY** be
```
Comment (regenscheid) [inline _FIPS201/frontend.md:700]:
```suggestion
present, the cryptographic operations that use the key management key **SHALL** be available through 
the contact interface and **MAY** additionally be available over the virtual contact interface of the PIV Card. Operations that use the key management key **SHALL NOT** be available through the contactless interface of the PIV Card. Private key operations **MAY** be
```

**Pull request merged**